### PR TITLE
Display the post author in the post meta informations

### PR DESCRIPTION
**Pull request by randruc**
# Summary

In config.toml, the key post_meta defines a range of meta informations that can
be displayed under the post title. This commit allows to display the author name
in these meta informations, by adding the constant "author" in the post_meta
range, like so:

post_meta = ["author", "date", "categories", "translations"]

This meta information will be displayed only if an author name is defined.
Moreover, if the Param authorbox is set to true in config.toml, the name is
displayed as a link to the authorbox on the same page.

When displaying post meta in the summary (from summary.html), the link is not created.
This trick is performed using the scratchpad. 

Let me know if more things are needed.

# Screenshots

With `post_meta = ["date", "categories", "translations"]`

image

With `post_meta = ["author", "date", "categories", "translations"]`

image

### CODE REVIEW

**Comment by randruc** [comment]
Hi!

I just noticed that the scratchpad trick doesn't work for some pages. I don't know why for know. I tell you if I have more informations about that bug.

**Comment by Vimux** [comment]
I have not done a code review yet, but I have a general question.

What is the reason for the change? I mean, Mainroad supports the only one author by default. So, this meta field will show identical author for all pages. We already have an author box for this.

When you have many authors, such meta field has a lot of meaning - It indicates the authors of each specific material plus you can go to the author’s page. In our case, the name is always the same, and the link is just an anchor. I don’t argue at all, this may be very useful for a specific site, but I have strong doubts about real usage in the theme.

**Comment by randruc** [comment]
I did more research on the bug this morning. The bug seems to be from the behaviour of the scratchpad. I will push a fix for that on my branch.

The big drawback I have with the authorbox is that it's on the bottom of the page. When the reader lands on a post page for the first time (s)he has no information at all about the author. I wish the author name could be visible immediately after the title. For now now the authorbox is the only way to let the reader know who is the author.

These feature is a proposal. We all have different uses of the same piece of software. I'm submitting that on pull requests because these days I made a few different modifications on my own fork of this theme, which I like and use, and I thought other users could be interested in it.

**Comment by randruc** [comment]
I fixed the bug by unsetting the scratch pad value after rendering post meta. When Hugo rendered the first post page, it didn't visited the summary.html partial yet.

**Comment by randruc** [comment]
For conflicts resolution, do you prefer I merge, or rebase then force push?

**Comment by Vimux** [comment]
Merge, please.

**Comment by randruc** [comment]
Thanks for replying. I will resolve tonight.

Do you have anything to say about the implementation? Maybe you are thinking about someting smarter than the scratchpad.

**Comment by randruc** [comment]
I just resolved the conflict on README.md, including last changes made on upstream master.

**Comment by randruc** [comment]
Hi. I integrated your suggestions. I agreed with you for the simplification of this logic and get rid of the creation of that link.
For the SVG icon, I picked a free one from the Font Awesome free collection. I've put the attribution in the project README.md file.

About the display of these post metas in the summary (eg. on the home page), I'm actually thinking about creating a new site param ('post_meta_summary') to configure which post metas to display on the home page.

Let me know about your new review.

**Comment by randruc** [comment]
Hi,

Sorry for the delay.
Here is the updated pull request, integrating all of your suggestions.

File: layouts/partials/post_meta/author.html [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{{- if .Site.Author.name -}}
+<div class="meta__item-author meta__item">
+	{{ partial "svg/author.svg" (dict "class" "meta__icon") -}}
+	<span class="meta__text">{{ .Site.Author.name }}</span>
+</div>
+{{- end -}}
```
Comment (Vimux) [inline layouts/partials/post_meta/author.html:6]: It looks like the last update accidentally added a newline here.

**Comment by randruc** [comment]
Hi,

You're right. I fixed it.

**Comment by randruc** [comment]
You're welcome. Thanks to you.

**Pull request merged**